compiler/src/dotty/tools/dotc/transform/Inlining.scala

```diff
@@ -33,11 +33,22 @@ import scala.annotation.constructorOnly
 /** Inlines all calls to inline methods that are not in an inline method or a quote */
 class Inlining extends MacroTransform {
   import tpd._
+  import Inlining._
 
   override def phaseName: String = Inlining.name
 
   override def allowsImplicitSearch: Boolean = true
 
+  override def run(using Context): Unit =
+    if ctx.compilationUnit.needsInlining then
+      try super.run
+      catch case _: CompilationUnit.SuspendException => ()
+
+  override def runOn(units: List[CompilationUnit])(using Context): List[CompilationUnit] =
+    val newUnits = super.runOn(units).filterNot(_.suspended)
```

Comment:
Is it `suspendedAtInliningPhase` instead of `suspended`? Or do we really need `suspendedAtInliningPhase`?

```suggestion
    val newUnits = super.runOn(units).filterNot(_. suspendedAtInliningPhase)
```

Reply:
No, it needs to be `.suspended` as we do not want to transform compilation units that were suspended in typer.

`suspendedAtInliningPhase` is only needed to let `sbt.ExtractDependencies` know that it has already registered the classes of this compilation unit.